### PR TITLE
Add normalised code column

### PR DESCRIPTION
**Pull request by SimonH**
This PR builds on https://github.com/trainline-eu/stations/pull/786, which should be merged first. (or ignored if this PR is approved as a single item)
Using the new same_as values to determine the correct value to add for a new normalised_code column. 

Currently the column is only set for non-GB locations. The GB locations will be updated separately at a later date.

I have added a unit test to ensure that all rows have the correct value set in the new column.

### CODE REVIEW

**Comment by noplay** [comment]
In your PR the normalized code look like: urn:trainline:public:nloc:eu18

I think the normalized code should not include eu in their name to be futur proof if one day we expand outside the European continent.

**Comment by SimonH** [comment]
Hi, the codes from atoc (which will be going against the GB stations in this file will be of the form

File: test_data.rb [hunk 1]
```diff
@@ -491,6 +491,21 @@ def test_main_station_hints
     end
   end
 
+  def test_normalised_code
+    STATIONS.each do |row|
+      if (row["country"] != 'GB' )
```
Comment (avandecreme) [inline test_data.rb]: We usually don't use parenthsis here in ruby.

File: test_data.rb [hunk 2]
```diff
@@ -491,6 +491,21 @@ def test_main_station_hints
...
+        assert !row["normalised_code"].nil?, "Station #{row["name"]} (#{row["id"]}) does not have a normalised_code"
+        expected_id = row["id"]
+        if (!row["same_as"].nil?)
```
Comment (avandecreme) [inline test_data.rb]: Same

File: test_data.rb (hunk 1 above)
Comment (SimonH) [reply test_data.rb]: Done. Thanks

File: test_data.rb [hunk 3]
```diff
@@ -491,6 +491,21 @@ def test_main_station_hints
...
+          expected_id = row["same_as"]
+        end
```
Comment (avandecreme) [inline test_data.rb]: `expected_id = row["same_as"] || row["id"]` seems clearer to me

File: test_data.rb (hunk 2 above)
Comment (SimonH) [reply test_data.rb]: Done

File: test_data.rb (hunk 3 above)
Comment (SimonH) [reply test_data.rb]: Done. Thanks. I'm completely new to Ruby

**Pull request merged**